java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -75,6 +75,28 @@ public static <T, U extends T> U[] castArray(T[] objectArr, Class<U> clazz) {
                 .toArray(size -> (U[]) Array.newInstance(clazz, size));
     }
 
+    /**
+     * Casts an array of arrays of objects to an array of arrays of type T.
+     *
+     * @param outerObjectArr Array of arrays of objects to cast.
+     * @param clazz The class of the array elements to cast to.
+     * @return An array of arrays of type U, containing the elements from the input array.
+     * @param <T> The base type from which the elements are being cast.
+     * @param <U> The subtype of T to which the elements are cast.
+     * @param <V> The array type of U
```

Comment:
`T` is always `Object`
`V` is `U[]`

I think you can leave here only one type parameter